src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -157,10 +157,17 @@ errors::eType TNumberSchemaItem<NumberType>::validate(
     report->set_validation_info(validation_info);
     return errors::INVALID_VALUE;
   }
+
   NumberType value(0);
   if (typeid(int32_t) == typeid(value)) {
+    if (Object.asInt() > std::numeric_limits<int32_t>::max() ||
+        Object.asInt() < std::numeric_limits<int32_t>::min())
+      return errors::OUT_OF_RANGE;
```

Comment:
@RomanReznichenkoLuxoft in [your case](https://google.github.io/styleguide/cppguide.html#Conditionals) you have to add braces for return statement